sanic/response.py

```diff
@@ -112,8 +114,16 @@ async def send(
         """
         if data is None and end_stream is None:
             end_stream = True
-        if end_stream and not data and self.stream.send is None:
-            return
+        if self.stream.send is None:
+            if end_stream and not data:
+                return
+            elif self.stream.stage == Stage.IDLE:
+                raise SanicException(
+                    "Response stream was ended, no more response data is"
+                    " allow ed to be sent."
```

Comment:
Extra space in `allow ed`.
Also, add the trailing space to the first line and remove from before `allow`.